sysid-application/src/main/native/cpp/view/Analyzer.cpp

```diff
@@ -86,21 +134,31 @@ void Analyzer::DisplayFileSelector() {
   ImGui::InputText("##location", &m_location, ImGuiInputTextFlags_ReadOnly);
 }
 
+void Analyzer::ResetData() {
+  m_plot.ResetData();
+  m_manager = std::make_unique<AnalysisManager>(m_settings, m_logger);
+  m_location = "";
+  m_ff = std::vector<double>{1, 1, 1};
+  UpdateFeedbackGains();
+}
+
 bool Analyzer::DisplayResetAndUnitOverride() {
+  auto type = m_manager->GetAnalysisType();
+  auto unit = m_manager->GetUnit();
+  auto factor = m_manager->GetFactor();
```

Comment:
What's the reasoning behind getting rid of the member variables?

Reply:
Duplicating state is never a good idea, because then you have to keep it synchronized.